UefiPayloadPkg/UefiPayloadEntry/FitUniversalPayloadEntry.c

```diff
@@ -554,8 +561,13 @@ FitUplEntryPoint (
     PrintHob (mHobList);
     );
 
-  FixUpPcdDatabase (DxeFv);
-  Status = UniversalLoadDxeCore (DxeFv, &DxeCoreEntryPoint);
+  //
+  // Fetch the FvHob for the DXE FV that we've now populated.
+  //
+  DxeFvHob = GetFirstHob (EFI_HOB_TYPE_FV);
+
+  FixUpPcdDatabase ((EFI_FIRMWARE_VOLUME_HEADER *)(UINTN)DxeFvHob->BaseAddress);
+  Status = UniversalLoadDxeCore ((EFI_FIRMWARE_VOLUME_HEADER *)(UINTN)DxeFvHob->BaseAddress, &DxeCoreEntryPoint);
   ASSERT_EFI_ERROR (Status);

```

Comment:
This patch will cause system assert in line 571 , due to get the wrong FV. Please help to check it.

Reply:
I don't understand: there are now two ways to hit `BuildFvHob ((EFI_PHYSICAL_ADDRESS)0, 0);` and they should be mutually exclusive. I preserved the previous behaviour. Can you attach a log?

One thing that comes to mind is that the UPL spec recently renamed `/options/upl-images@<...>` to `/options/upl-image@<...>`, see https://github.com/UniversalPayload/spec/pull/18. UefiPayload has not yet followed suit. If you've applied that change to your Platform Init, it might be broken for this reason. I'll be pushing an update to this patch soon.